crates/api/src/config.rs

```diff
@@ -61,6 +61,19 @@ impl Config {
         Ok(())
     }
 
+    /// Update a module config with new values. Note, this will overwrite
+    /// ALL config values. If you would like to only update some, please
+    /// first call [Self::get_module_config], update the appropriate values,
+    /// then call this function again to set them.
+    pub fn set_module_config<M: ModConfig>(
```

Comment:
This means that it's going to be tricky to report which config items that have been changed don't have any effect when changed at runtime. It might be easier to receive patch/update request when dealing with that kind of thing?

Reply:
I'm not sure I fully understand your suggestion, although I'm also not super happy with the interface here.

My initial thought was to leave the structure completely up to the module implementations, but perhaps this would all be more intuitive if we exposed a map of properties for each module in the interface. Then only those properties that were included in the set call would be overwritten?

Reply:
I'm thinking if I'm a module receiving new configuration and I see:

```
mymod.config1 = "hello",
mymod.config2 = 20,
```

Then I get an update saying:


```
mymod.config1 = "hallo",
mymod.config2 = 25,
```

Then I have to figure out that `mymod.config1` changed which let's say isn't supported. Then I also have to figure out that `mymod.config2` changed and potentially take some action to start using that new config.

If instead I only get an input that says:


```
mymod.config2 = 25,
```

Then I can easily check that only properties that are allowed to be changed are changing and I can take an actions required on other config values changing as they're applied to my config set.

Reply:
This is not for runtime changes in config. It all happens before the module receives the config in the constructor. Actually, this could be considered a test function, since in a production use-case this should all be set via config file. I only needed this so I could alter the defaults for specific tests.

But this is certainly pointing to a need for the api to be different : )

Reply:
Ahh okay, so can/should this go behind `#[cfg(test)]` for now?

Reply:
I'm working on a new config strategy. If it works out we can merge that first and I'll update this code to use it and hopefully it won't be as awkward.

Reply:
Okay, please see this: https://github.com/holochain/kitsune2/pull/58